Client/Frontend/Settings/HomePageSettingViewController.swift

```diff
@@ -32,21 +32,25 @@ class HomePageSettingViewController: SettingsTableViewController {
         let showTopSites = CheckmarkSetting(title: NSAttributedString(string: Strings.SettingsNewTabTopSites), subtitle: nil, accessibilityIdentifier: nil, isEnabled: {return self.currentChoice == NewTabPage.topSites}, onChanged: {
```

Comment:
nit: `func finishOnChanged() { <common code> }` above this line, then in onChanged 
```
self.currentChoice = ...
finishOnChanged()
```